File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder(self, root, h, k):
        if not root:
            return
        self.inorder(root.left,h, k)
        k[0] = k[0] - 1
        if k[0] == 0:
            h.append(root.val)
        self.inorder(root.right,h, k)
            

    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        h = []
        g = [k]
        self.inorder(root, h,g)
        return h[0]

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void inorder(TreeNode* root, int* k, int* ans){
        if(root == nullptr){
            return;
        }
        inorder(root->left, k, ans);
        *k = *k - 1;
        if(*k == 0){
            *ans = root->val;
            return;        
        }
        inorder(root->right, k, ans);
        
    }
    int kthSmallest(TreeNode* root, int k) {
        int ans = 0;
        int j = k;
        inorder(root, &j, &ans);
        return ans;
    }
};